src/threads/scriptthread.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2014-2020 Patrizio Bekerle -- <patrizio@bekerle.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+
+#include "scriptthread.h"
+#include <QDebug>
+
+QMap<QString, int> ScriptThread::threadCounter;
+
+ScriptThread::ScriptThread(ScriptingService *ss,
+                           const TerminalCmd &cmd,
+                           const QString &identifier,
+                           const int index) : QThread() {
+    this->cmd = cmd;
+    this->identifier = identifier;
+    this->index = index;
+    increaseThreadCounter();
+    ScriptThread::connect(this, SIGNAL(callBack(ScriptThread*)), ss, SLOT(onScriptThreadDone(ScriptThread*)));
+}
+
+void ScriptThread::increaseThreadCounter(){
```

Comment:
please use a white-space before the {

Reply:
Done. Is there a specific reason for this or just a convention?

Reply:
Convention, please generally try to mimic the current code style (as long as it does not contradict itself) :grin:

Reply:
Sure